core/src/test/java/org/bitcoinj/core/FloLegacyAddressTest.java

```diff
@@ -0,0 +1,271 @@
+import org.bitcoinj.params.MainNetParams;
+import org.bitcoinj.params.Networks;
+import org.bitcoinj.params.TestNet3Params;
+import org.bitcoinj.script.Script;
+import org.bitcoinj.script.ScriptBuilder;
+import org.bitcoinj.script.ScriptPattern;
+import org.bitcoinj.script.Script.ScriptType;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.bitcoinj.core.Utils.HEX;
+import static org.junit.Assert.*;
+
+/**
+ * Copied from LegacyAddressTest
+ */
+public class FloLegacyAddressTest {
+    private static final NetworkParameters FLOTESTNET = FloTestNetParams.get();
+    private static final NetworkParameters FLOMAINNET = FloMainNetParams.get();
+
+    @Test
+    public void testJavaSerialization() throws Exception {
+        LegacyAddress testAddress = LegacyAddress.fromBase58(FLOTESTNET, "oYRaaENhPh4BM3Ync7xACL7z2LxMxA5z2R");
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        new ObjectOutputStream(os).writeObject(testAddress);
+        LegacyAddress testAddressCopy = (LegacyAddress) new ObjectInputStream(new ByteArrayInputStream(os.toByteArray()))
+                .readObject();
+        assertEquals(testAddress, testAddressCopy);
+
+        LegacyAddress mainAddress = LegacyAddress.fromBase58(FLOMAINNET, "F8wso89xU3FhBVkBsZc6RNdA1x8YFCE6xi");
+        os = new ByteArrayOutputStream();
+        new ObjectOutputStream(os).writeObject(mainAddress);
+        LegacyAddress mainAddressCopy = (LegacyAddress) new ObjectInputStream(new ByteArrayInputStream(os.toByteArray()))
+                .readObject();
+        assertEquals(mainAddress, mainAddressCopy);
+    }
+
+    @Test
+    public void stringification() throws Exception {
+        // Test a testnet address.
+        LegacyAddress a = LegacyAddress.fromPubKeyHash(FLOTESTNET, HEX.decode("aaa17faff597a3a5d20ba5564b2e7104cf2d62ed"));
+        assertEquals("oYRaaENhPh4BM3Ync7xACL7z2LxMxA5z2R", a.toString());
+        assertEquals(ScriptType.P2PKH, a.getOutputScriptType());
+
+        LegacyAddress b = LegacyAddress.fromPubKeyHash(FLOMAINNET, HEX.decode("222fe16d40ff72c2337d07748277d42d8cd0ce2f"));
+        assertEquals("F8wso89xU3FhBVkBsZc6RNdA1x8YFCE6xi", b.toString());
+        assertEquals(ScriptType.P2PKH, a.getOutputScriptType());
+    }
+    
+    @Test
+    public void decoding() throws Exception {
+        LegacyAddress a = LegacyAddress.fromBase58(FLOTESTNET, "oYRaaENhPh4BM3Ync7xACL7z2LxMxA5z2R");
+        assertEquals("aaa17faff597a3a5d20ba5564b2e7104cf2d62ed", Utils.HEX.encode(a.getHash()));
+
+        LegacyAddress b = LegacyAddress.fromBase58(FLOMAINNET, "F8wso89xU3FhBVkBsZc6RNdA1x8YFCE6xi");
+        assertEquals("222fe16d40ff72c2337d07748277d42d8cd0ce2f", Utils.HEX.encode(b.getHash()));
+    }
+    
+    @Test
+    public void errorPaths() {
```

Comment:
nitpick: this would be easier to read if you split each scenario into its own method, and on each method you can change the @Test annotation to say which exception is expected to be thrown by the method body.

if you want to keep this the same to mirror the original testcase, then feel free to ignore this

Reply:
I'm currently on the side of keeping things the same... I feel there's consolidation if we keep building out coin-specific tests, and that task will be easier the less we diverge.